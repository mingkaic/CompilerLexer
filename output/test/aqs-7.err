output prefix: llvm/test/aqs-7
llvmas: /Users/cmk/llvm/bin/llvm-as
llc: /Users/cmk/llvm/bin/llc
cc: gcc
codegen: /Users/cmk/Developer/cworkspace/compiler/hw4/Lexer/answer/decafcomp
stdlib: /Users/cmk/Developer/cworkspace/compiler/hw4/Lexer/answer/decaf-stdlib.c
generating llvm code:/Users/cmk/Developer/cworkspace/compiler/hw4/Lexer/answer/decafcomp ... ok
; ModuleID = 'QuickSort'

define void @quickSort(i32 %left, i32 %right) {
entry:
  %right2 = alloca i32
  %left1 = alloca i32
  store i32 %left, i32* %left1
  store i32 %right, i32* %right2
  %right3 = load i32, i32* %right2
  %left4 = load i32, i32* %left1
  %subtmp = sub i32 %right3, %left4
  %letmp = icmp ule i32 %subtmp, 0
  br i1 %letmp, label %then, label %else

then:                                             ; preds = %entry
  ret void
}
assembling to bitcode:/Users/cmk/llvm/bin/llvm-as "llvm/test/aqs-7.llvm" -o "llvm/test/aqs-7.llvm.bc" ... failed (1)
/Users/cmk/llvm/bin/llvm-as: llvm/test/aqs-7.llvm:13:36: error: use of undefined value '%else'
  br i1 %letmp, label %then, label %else
                                   ^
converting to native code:/Users/cmk/llvm/bin/llc "llvm/test/aqs-7.llvm.bc" -o "llvm/test/aqs-7.llvm.s" ... ok
linking:gcc -o "llvm/test/aqs-7.llvm.exec" "llvm/test/aqs-7.llvm.s" "/Users/cmk/Developer/cworkspace/compiler/hw4/Lexer/answer/decaf-stdlib.c" ... failed (1)
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
running:llvm/test/aqs-7.llvm.exec ...Traceback (most recent call last):
  File "/Users/cmk/Developer/cworkspace/compiler/hw4/Lexer/answer/llvm-run", line 172, in <module>
    result &= run("running", "%s.llvm.exec" % (out_prefix), ".run", None, out_prefix)
  File "/Users/cmk/Developer/cworkspace/compiler/hw4/Lexer/answer/llvm-run", line 90, in run
    retval = subprocess.call(shlex.split(cmd), stdin=infile, stdout=outoutfile, stderr=outerrfile)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 522, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 710, in __init__
    errread, errwrite)
  File "/usr/local/Cellar/python/2.7.10_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 1335, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
