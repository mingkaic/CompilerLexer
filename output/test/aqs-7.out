Program(None,Package(QuickSort,None,Method(quickSort,VoidType,VarDef(left,IntType),VarDef(right,IntType),MethodBlock(None,IfStmt(BinaryExpr(Minus,VariableExpr(right),BinaryExpr(Leq,VariableExpr(left),NumberExpr(0))),Block(None,BinaryExpr(Leq,VariableExpr(left),NumberExpr(0))),Block(VarDef(part,IntType),VarDef(pivot,IntType),AssignVar(pivot,ArrayLocExpr(list,VariableExpr(right))),AssignVar(part,MethodCall(partition,VariableExpr(left),VariableExpr(right),VariableExpr(pivot))),MethodCall(quickSort,VariableExpr(left),BinaryExpr(Minus,VariableExpr(part),NumberExpr(1))),MethodCall(quickSort,BinaryExpr(Plus,VariableExpr(part),NumberExpr(1)),VariableExpr(right))))))))
